From 9fab5256d6a1edc536c58f24abe3626dd17f4cdc Mon Sep 17 00:00:00 2001
From: cedricve <hello@cedric.ws>
Date: Tue, 29 Mar 2016 14:05:19 -0400
Subject: [PATCH 2/2] change session cache

Signed-off-by: cedricve <hello@cedric.ws>
---
 app/config/session.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/config/session.php b/app/config/session.php
index 4cf30f6..908064e 100755
--- a/app/config/session.php
+++ b/app/config/session.php
@@ -16,7 +16,7 @@ return array(
 	|
 	*/
 
-	'driver' => 'lockfile',
+	'driver' => 'memcached',
 
 	/*
 	|--------------------------------------------------------------------------
-- 
1.9.1

